sphinx/texinputs/footnotehyper-sphinx.sty

```diff
@@ -87,7 +102,7 @@
 \def\FNH@endfntext@nolink {\begingroup
     \let\@makefntext\@empty\let\@finalstrut\@gobble
     \let\rule\@gobbletwo
-    \if@savingnotes\expandafter\fn@fntext\else\expandafter\H@@footnotetext\fi
+    \if@savingnotes\expandafter\FNH@fn@fntext\else\expandafter\H@@footnotetext\fi
```

Comment:
The `\H@@footnotetext` macro is the original `\@footnotetext` that `hyperref` keeps in addition to the `\@footnotetext` patched for hyperlinks. The `\FNH@fn@fntext` is the one which is used inside `savenotes` environment, it is the original `\fn@fntext` from footnote.sty package. But the`\fn@fntext` in earlier version referred to the redefined one by footnotehyper-sphinx, `\FNH@hyper@fntext` hence it was a bug here but escaped detection because Sphinx inserted all `footnotetext` environment after the tabular, tabulary, threeparttable, hence the footnote version of footnotehyper-sphinx was not executed, and the bug did not show. But if one starts locating the footnotetext inside the table, then the bug shows. This line fixes it.